cmd/ingester/app/flags.go

```diff
@@ -106,14 +108,61 @@ func AddFlags(flagSet *flag.FlagSet) {
 		DefaultDeadlockInterval,
 		"Interval to check for deadlocks. If no messages gets processed in given time, ingester app will exit. Value of 0 disables deadlock check.")
 
+	// TODO: Remove deprecated flags after 1.11
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixBrokers,
+		"",
+		fmt.Sprintf("Deprecated; replaced by %s", KafkaConsumerConfigPrefix+SuffixBrokers))
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixTopic,
+		"",
+		fmt.Sprintf("Deprecated; replaced by %s", KafkaConsumerConfigPrefix+SuffixTopic))
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixGroupID,
+		"",
+		fmt.Sprintf("Deprecated; replaced by %s", KafkaConsumerConfigPrefix+SuffixGroupID))
+	flagSet.String(
+		DeprecatedKafkaConfigPrefix+SuffixEncoding,
+		"",
+		fmt.Sprintf("Deprecated; replaced by %s", KafkaConsumerConfigPrefix+SuffixEncoding))
 }
 
 // InitFromViper initializes Builder with properties from viper
 func (o *Options) InitFromViper(v *viper.Viper) {
-	o.Brokers = strings.Split(v.GetString(KafkaConfigPrefix+SuffixBrokers), ",")
-	o.Topic = v.GetString(KafkaConfigPrefix + SuffixTopic)
-	o.GroupID = v.GetString(KafkaConfigPrefix + SuffixGroupID)
-	o.Encoding = v.GetString(KafkaConfigPrefix + SuffixEncoding)
+	o.Brokers = strings.Split(v.GetString(KafkaConsumerConfigPrefix+SuffixBrokers), ",")
```

Comment:
we recently fixed a bug where the list of hosts for Cassandra or ES would not parse correctly if they contained spaces. Perhaps worth reusing the same util method.

Reply:
I’ll bundle that suggestion into the PR tha delete the deprecated flags (to be merged post 1.11)